src/wavm/wavm.cc

```diff
@@ -295,38 +275,26 @@ std::unique_ptr<WasmVm> Wavm::clone() {
   return wavm;
 }
 
-bool Wavm::load(const std::string &code, bool allow_precompiled) {
+bool Wavm::load(std::string_view code, bool is_precompiled,
+                std::unordered_map<uint32_t, std::string>) {
   ASSERT(!has_instantiated_module_);
   has_instantiated_module_ = true;
   compartment_ = WAVM::Runtime::createCompartment();
   context_ = WAVM::Runtime::createContext(compartment_);
-  if (!loadModule(code, ir_module_)) {
-    return false;
-  }
 
-  // Get ABI version from bytecode.
-  if (!common::BytecodeUtil::getAbiVersion(code, abi_version_)) {
-    fail(FailState::UnableToInitializeCode, "Failed to parse corrupted Wasm module");
-    return false;
-  }
-
-  std::string_view precompiled = {};
-  if (allow_precompiled) {
-    if (!common::BytecodeUtil::getCustomSection(code, getPrecompiledSectionName(), precompiled)) {
-      fail(FailState::UnableToInitializeCode, "Failed to parse corrupted Wasm module");
+  if (is_precompiled) {
+    module_ =
+        WAVM::Runtime::loadPrecompiledModule(ir_module_, {code.data(), code.data() + code.size()});
+  } else {
+    if (!WASM::loadBinaryModule(reinterpret_cast<const unsigned char *>(code.data()), code.size(),
+                                ir_module_)) {
       return false;
```

Comment:
Actually, it looks that WAVM needs both: the bytecode and the precompiled module.

I'll get this fixed, but it changes the API a bit.